### PR TITLE
feat: compute reservation create

### PR DESCRIPTION
**Pull request by gryczj**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by subfuzion — CHANGES_REQUESTED** [review]
Hi @gryczj. Last week I helped with a [few changes](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3800/files#diff-8064cb44676aa7d64757406dcbba1407ab68274b5cb53aba1ca09381ac29864eR36-R113) to try to stabilize ensuring resources didn't exist from previous failed tests and were removed after tests. I think you need to work with your team to come up with a more optimal solution for handling this cleanly and reliably. It seems that this is a problematic area (I had to insert delays with retries, but it felt like a bit of a brute force hack). Perhaps for compute instances, you really do need to separate these concerns into a more robust set up and tear down module with retries that you import that ensures this happens in a consistently reliable way before erroring. If it's this hard for us to do, then it's going to be hard for our customers as well, so let's convert this to draft mode while you take a bit of time to coordinate with your team to come up with something that showcases the proper way to handle this. Thank you!

**Comment by gryczj** [comment]
> Hi @gryczj. Last week I helped with a [few changes](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3800/files#diff-8064cb44676aa7d64757406dcbba1407ab68274b5cb53aba1ca09381ac29864eR36-R113) to try to stabilize ensuring resources didn't exist from previous failed tests and were removed after tests. I think you need to work with your team to come up with a more optimal solution for handling this cleanly and reliably. It seems that this is a problematic area (I had to insert delays with retries, but it felt like a bit of a brute force hack). Perhaps for compute instances, you really do need to separate these concerns into a more robust set up and tear down module with retries that you import that ensures this happens in a consistently reliable way before erroring. If it's this hard for us to do, then it's going to be hard for our customers as well, so let's convert this to draft mode while you take a bit of time to coordinate with your team to come up with something that showcases the proper way to handle this. Thank you!

@subfuzion Please see: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3808

File: compute/test/createComputeHyperdiskFromPool.test.js [hunk 1]
```diff
@@ -18,14 +18,14 @@
 
 const path = require('path');
 const {assert} = require('chai');
-const {after, before, describe, it} = require('mocha');
+const {after, before, xdescribe, it} = require('mocha');
```
Comment (gryczj) [inline compute/test/createComputeHyperdiskFromPool.test.js:21]: Skip until https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3808 is not resolved

**Comment by gryczj** [comment]
Please review https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3808 first

**Review by subfuzion — APPROVED** [review]

**Comment by subfuzion** [comment]
@gryczj I merged #3808 and then updated this branch, but that doesn't seem to have resolved the issue, unfortunately:

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/actions/runs/10686130157/job/29620561886?pr=3807#step:9:268

File: compute/disks/createComputeHyperdiskPool.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ async function main() {
   // Project ID or project number of the Google Cloud project you want to use.
   const projectId = await storagePoolClient.getProjectId();
   // Name of the zone in which you want to create the storagePool.
-  const zone = 'europe-central2-b';
+  const zone = 'us-central1-a';
```
Comment (gryczj) [inline compute/disks/createComputeHyperdiskPool.js:36]: We can't create storage pool in europe zone. Changed to US as it was before. The same problem was in Java directory: https://github.com/GoogleCloudPlatform/java-docs-samples/commit/e24337003c62b37f3894214849d6c38d9d831381

**Pull request merged**